File: App1/app/src/main/java/com/example/user/app1/MainActivity.java
package com.example.user.app1;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button button1;
    Button button2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        button1 = findViewById(R.id.button1);
        button2 = findViewById(R.id.button2);
        button1.setOnClickListener(oncBtnOk1);
        button2.setOnClickListener(oncBtnOk2);
    }

    View.OnClickListener oncBtnOk1 = new View.OnClickListener() {
        @Override
        public void onClick(View view) {
            startActivity(new Intent(MainActivity.this, Activity1.class));
        }
    };

    View.OnClickListener oncBtnOk2 = new View.OnClickListener() {
        @Override
        public void onClick(View view) {
            startActivity(new Intent(MainActivity.this, Activity2.class));
        }
    };
}

File: App2/app/src/main/java/com/example/a1/app2/MainActivity.kt
package com.example.a1.app2

import android.content.Context
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    companion object {

        fun createStartIntentAct1(context: Context): Intent {
            return Intent(context, Activity1::class.java)
        }

        fun createStartIntentAct2(context: Context): Intent {
            return Intent(context, Activity2::class.java)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        button1.setOnClickListener { startActivity(createStartIntentAct1(this)) }
        button2.setOnClickListener { startActivity(createStartIntentAct2(this)) }
    }
}
